### ISSUE TITLE
Problem writing dataframe with one column

### ISSUE THREAD

**Issue by Riqardos** [aws/amazon-redshift-python-driver#197]
## Redshift version
PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.58814\x00

## Client Operating System
MacOS 14.1 (23B74)

## Python version
3.11

## Table schema
Simple table with one column.

## Problem description
Have problem running `write_dataframe()` with dataframe having only one column. Dataframe with two columns works as expected.

Not working
```
import redshift_connector
conn = redshift_connector.connect(...)
cursor = conn.cursor()
conn.autocommit = True
import pandas as pd
cursor.execute("""
    CREATE TABLE etl.debug_table (
        sum INT
    )
""")

df_one_row = pd.DataFrame({'sum': [1]})
cursor = redshift_etl.conn.cursor()
cursor.write_dataframe(df_one_row, 'etl.debug_table')
```

## Python Driver trace logs
```

KeyError                                  Traceback (most recent call last)
File [~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1586](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1586), in Connection.make_params(self, values)
   [1585](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1585) try:
-> [1586](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1586)     params.append(self.py_types[typ])
   [1587](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1587) except KeyError:

KeyError: <class 'numpy.int64'>

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
File [~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1589](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1589), in Connection.make_params(self, values)
   [1588](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1588) try:
-> [1589](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1589)     params.append(self.inspect_funcs[typ](value))
   [1590](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/core.py:1590) except KeyError as e:

KeyError: <class 'numpy.int64'>

During handling of the above exception, another exception occurred:

NotSupportedError                         Traceback (most recent call last)
File [~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:598](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:598), in Cursor.write_dataframe(self, df, table)
    [597](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:597) if len(arrays) == 1:
--> [598](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:598)     self.execute(sql, arrays[0])
...
    [604](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:604)     )
    [605](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:605) finally:
    [606](https://file+.vscode-resource.vscode-cdn.net/Users/riqardos/jobs/python/~/Library/Caches/pypoetry/virtualenvs/etl-CfOdF0rB-py3.11/lib/python3.11/site-packages/redshift_connector/cursor.py:606)     self.paramstyle = cursor_paramstyle

InterfaceError: An error occurred when attempting to insert the pandas.DataFrame into $etl.debug_table
```

## Reproduction code
```
import redshift_connector
import pandas as pd

conn = redshift_connector.connect(...)
cursor = conn.cursor()
conn.autocommit = True

cursor.execute("""
    CREATE TABLE etl.debug_table (
        sum INT
    )
""")

df_one_row = pd.DataFrame({'sum': [1]})
cursor = redshift_etl.conn.cursor()
cursor.write_dataframe(df_one_row, 'etl.debug_table')
```

**Comment by Brooke-white** [comment]
Hi @Riqardos , thank you for reporting this issue. The team will work on reproducing and determining resolution steps.

**Comment by Brooke-white** [comment]
Hi @Riqardos, I've been able to reproduce this bug and have identified the root cause. In `write_dataframe()` we directly pass the numpy representation of the DataFrame to the `cursor.execute()` method. When preparing the bind parameters (i.e. the numpy representation of the DataFrame) to send to Redshift, we fail to find a mapping for `numpy.int64` in our internal type conversion dictionary. What this means is that redshift-connector does not know how to convert `numpy.int64` into a format understandable by Redshift server, so it throws as error. We do not provide mappings for any `numpy` datatypes in our type conversion dictionaries as they simply handle mappings between Python and Redshift types.

This issue had not been detected earlier as our integration tests for this method only test with Python strings. If you change the reproduction code you provided to `df_one_row = pd.DataFrame({'sum': ["1"]})` the code executes successfully, but of course that's not a proper solution to this.

I'm thinking a quick solution to this issue is to convert the underlying numpy.ndarray within the pandas.DataFrame to a list using `tolist()`. [tolist()](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html) handles the conversion to Python types which is likely a better place for that conversion to happen rather than in redshift-connector.

I'll need to perform some more testing for this approach, since the current testing clearly lacks cases where the DataFrame contains numpy data-types, but anticipate we can include this in our December release.

**Comment by Brooke-white** [comment]
Hi @Riqardos , a fix is present in 2.0.918
